src/blog/0.3/linear-types-borrowing.vmd

```diff
@@ -0,0 +1,673 @@
+
+
+All of the above drawbacks exist because neither linear typing nor borrow checking can have *shared mutability*. In other words, multiple mutable references to the same data.
+
+
+<slice/>
+
+# Addressing the Drawbacks
+
+Are there any ways we can address those downsides without sacrificing the benefits?
+
+
+In other words, is there a way we can have linear types _and_ shared mutability?
+
+
+Adding borrow checking partially helps with drawback #1 [# Partially because borrow checking lets us have temporary references, not persistent ones. The borrow checker has a hard time putting references into data structures.] but it still has drawbacks #2, #3, and #4. Rust's `Rc` and `RefCell` helps with those, but they're incompatible with linear types' benefits [# We can't put a linear type into an `Rc` because `Rc` throws away its contents when the last alias disappears.], and have their own drawbacks (like forcing heap allocation and not having [higher RAII](https://verdagon.dev/blog/higher-raii-7drl)).
+
+
+[# E: How is this next part, where we blend gen refs back in? I tried to find an elegant way to build up to "the solution was with us all along".]
+
+The other solution, surprisingly, is to blend generational references back in.
+
+Each object is still owned linearly; only one place can own an object at a time. But we'll also allow a generational reference to point at the object.
+
+With this, we can address a lot of the drawbacks:
+
+ 1. We can have as many references to data as we want, though there is still only one owner.
+ 1. We don't need to restructure our program to accommodate them.
+ 1. We can use patterns like intrusive data structures and graphs. [# In Vale, we would still have all nodes owned by a central collection, but we could still have references between nodes just fine.]
+ 1. We can use observers, back-references, dependency references, callbacks, delegates, whatever we want.
+ 1. We still don't have data races, because Vale changes the generations of any objects that cross region boundaries (unless we safely avoid that with [isolates](/blog/zero-cost-borrowing-regions-part-2-isolates)).
+
+
+Generational references seem to occupy a sweet spot because they allow objects to be linear (they're owned by only one place), yet allow shared mutability in a way that doesn't artificially extend the lifetime of the object.
+
+
+Since we're combining linear types with aliasing, we'll call this the *linear-aliasing model*.
+
+
+The mark of a good language is when its features compose well and don't conflict with each other. It seems we're on the right track with this model, as it composes linear types with shared mutability pretty well.
```

Comment:
"pretty well."
You're just restating your definition, maybe define what 'pretty well' means here, like "cohesively and without adding a ton of edge cases"